packages/utils/src/lib/types.ts

```diff
@@ -15,3 +15,10 @@ export type WithRequired<T, K extends keyof T> = Prettify<
 >;
 
 export type Prettify<T> = { [K in keyof T]: T[K] };
+
+export type CamelCaseToKebabCase<T extends string> =
+  T extends `${infer First}${infer Rest}`
+    ? Rest extends Uncapitalize<Rest>
+      ? `${Lowercase<First>}${CamelCaseToKebabCase<Rest>}`
+      : `${Lowercase<First>}-${CamelCaseToKebabCase<Rest>}`
+    : T;
```

Comment:
The inferred type doesn't match the returned string from `camelCaseToKebabCase`:

![image](https://github.com/user-attachments/assets/b5157041-3699-4164-b72a-9370d29bcf1c)

It's important (for developer sanity) that the _type is consistent with reality_, even if that means making the function less "robust". For example, the `'hello_world test'` input string isn't in camelCase, so converting it need not be the function's responsibility. After all, the function is called `camelCaseToKebabCase`, it doesn't claim to convert _anything_ to kebab-case. Simplifying the runtime conversion will also simplify the type conversion - reimplementing the current conversion in types would very challenging otherwise.

Reply:
BTW, Vitest supports [Testing Types](https://vitest.dev/guide/testing-types). We should consider that for these complex types. I actually used it recently [here](https://gitlab.com/code-pushup/gitlab-pipelines-template/-/blob/main/src/env.test-d.ts?ref_type=heads) and it works really well, all I had to do was add `--typecheck` to `vitest` command. We could leave that for another PR, though.

Reply:
Fixed + added type tests